### PR TITLE
Fix for Issue #250

### PR DESCRIPTION
**Pull request by mariaschuld**
As discussed, I made the following changes:

- make seed default argument a fixed integer
- add 2 unit tests to check that two different seeds create different circuit result, while same integers or no explicit keyword argument creates two circuit with same result
- Explain use in docs via a note and warning message.

This should only affect users who use no specific keyword argument and expect the circuit to be different across two qnodes. But these users should be the ones having the bug of the circuit being rebuilt in every evaluation in the first place.

### CODE REVIEW

File: pennylane/templates/layers.py [hunk 1]
```diff
@@ -176,6 +178,15 @@ def RandomLayer(weights, wires, ratio_imprim=0.3, imprimitive=CNOT, rotations=No
         :width: 60%
         :target: javascript:void(0);
 
+    .. note::
+        Using the default seed (or any other fixed integer seed) generates one and the same circuit in every
+        quantum node. To generate different circuit architectures, either use a random seed, or use ``seed=None``
```
Comment (co9olguy) [inline pennylane/templates/layers.py]:
```suggestion
        quantum node. To generate different circuit architectures, either use a different random seed, or use ``seed=None``
```

File: tests/test_templates_layers.py [hunk 2]
```diff
@@ -548,11 +548,10 @@ def impr(self, request):
     def rots(self, request):
         return request.param
 
-    def test_random_layers_seed(self, n_layers, tol, seed):
-        """Test that pennylane.templates.layers.RandomLayers() gets deterministic when using seed."""
+    def test_random_layers_seed_deterministic(self, n_layers, tol, seed):
+        """Test that pennylane.templates.layers.RandomLayers() acts deterministically when using seed."""
```
Comment (co9olguy) [inline tests/test_templates_layers.py]:
```suggestion
        """Test that pennylane.templates.layers.RandomLayers() acts deterministically when using fixed seed."""
```

File: tests/test_templates_layers.py [hunk 3]
```diff
@@ -548,11 +548,10 @@ def impr(self, request):
     def rots(self, request):
         return request.param
 
-    def test_random_layers_seed(self, n_layers, tol, seed):
-        """Test that pennylane.templates.layers.RandomLayers() gets deterministic when using seed."""
+    def test_random_layers_seed_deterministic(self, n_layers, tol, seed):
```
Comment (co9olguy) [inline tests/test_templates_layers.py]: worthwhile to use `pytest.mark.parameterize` with a couple different possible seed values?

Comment (josh146) [reply tests/test_templates_layers.py]: :+1: good idea

Comment (mariaschuld) [reply tests/test_templates_layers.py]: The parametrised fixture "seed" should try different seeds [1,2,3] already. Or are you thinking of trying different seeds and their output value in a given circuit? 

Shall I rewrite the test or add another one?

Comment (mariaschuld) [reply tests/test_templates_layers.py]: It is not clear to me what `mark.parametrize` does that a parametrised fixture does not...

Comment (co9olguy) [reply tests/test_templates_layers.py]: Sounds like you were already on the same page. I didn't realize that `seed` was already parameterized. :+1:

File: pennylane/templates/layers.py [hunk 4]
```diff
@@ -162,7 +164,7 @@ def RandomLayers(weights, wires, ratio_imprim=0.3, imprimitive=CNOT, rotations=N
                     seed=seed)
 
 
-def RandomLayer(weights, wires, ratio_imprim=0.3, imprimitive=CNOT, rotations=None, seed=None):
+def RandomLayer(weights, wires, ratio_imprim=0.3, imprimitive=CNOT, rotations=None, seed=42):
```
Comment (zzzeid) [inline pennylane/templates/layers.py]: Do we expect users to ever pass `None` as a value here? There's a line below (195|206) checking if `seed is not None` which may not serve its original purpose any more (i.e. we may want to raise an exception or let the seeding fail if `None` is passed.)

Comment (josh146) [reply pennylane/templates/layers.py]: There is still a use-case for setting seed to `None` (see the note below)

**Pull request merged**